contracts/andromeda_vault/src/contract.rs

```diff
@@ -0,0 +1,426 @@
+            ado_type: "vault".to_string(),
+            operators: msg.operators,
+            modules: None,
+            primitive_contract: None,
+        },
+    )
+}
+
+#[entry_point]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::AndrReceive(msg) => execute_andr_receive(deps, env, info, msg),
+        ExecuteMsg::Deposit {
+            recipient,
+            amount,
+            strategy,
+        } => execute_deposit(deps, env, info, amount, recipient, strategy),
+        ExecuteMsg::Withdraw {
+            recipient,
+            withdrawals,
+            strategy,
+        } => execute_withdraw(deps, env, info, recipient, withdrawals, strategy),
+    }
+}
+
+fn execute_andr_receive(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: AndromedaMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        AndromedaMsg::Receive(data) => {
+            let strategy: Option<StrategyType> = parse_message(&data)?;
+            let sender = info.sender.to_string();
+            execute_deposit(
+                deps,
+                env,
+                info,
+                None,
+                Some(Recipient::Addr(sender)),
+                strategy,
+            )
+        }
+        _ => ADOContract::default().execute(deps, env, info, msg, execute),
+    }
+}
+
+fn execute_deposit(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    amount: Option<Coin>,
+    recipient: Option<Recipient>,
+    strategy: Option<StrategyType>,
+) -> Result<Response, ContractError> {
+    require(
+        !info.funds.is_empty()
+            && (amount.is_none()
+                || (amount.is_some() && amount.clone().unwrap().amount.gt(&Uint128::zero()))),
+        ContractError::InsufficientFunds {},
+    )?;
+    let mut resp = Response::default();
+    let recipient = recipient.unwrap_or_else(|| Recipient::Addr(info.sender.to_string()));
+
+    // If no amount is provided then the sent funds are used as a deposit
+    let deposited_funds = if let Some(deposit_amount) = amount {
+        let mut deposit_balance = Uint128::zero();
+        for funds in info.funds {
+            // Find funds in sent funds and add to deposit amount
+            if funds.denom == deposit_amount.denom {
+                deposit_balance = deposit_balance.checked_add(funds.amount)?;
+            }
+        }
+
+        // If depositing to a yield strategy we must first check that the sender has either provided the amount to deposit or has a combination of the amount within the vault and within the sent funds
+        if strategy.is_some() && deposit_balance <= deposit_amount.amount {
+            let vault_balance = BALANCES
+                .may_load(
+                    deps.storage,
+                    (
+                        recipient.get_addr(deps.api, &deps.querier, None).unwrap(),
```

Comment:
Should replace `unwrap` with `?`. How come we aren't passing in a `mission_contract`?

Reply:
Just a mistake on my end, thought this was in tests, well spotted.